### ISSUE TITLE
channelId parameter should be omitted when not needed 

### ISSUE THREAD

**Issue by Strixx76** [user2684/imouapi#10]
If channelId parameter is omitted in `getDeviceCameraStatus` and `setDeviceCameraStatus` when for `enableType` with scope `device`.
This will fix issues reported in https://github.com/user2684/imou_life/issues/82

**Comment by user2684** [comment]
Good point, channelId  usage has never be clear from the API documentation. I wonder if the fact that it applies to some cameras but not to others may be related to the fact that some cameras have multiple channels, most of the cameras only have channel 0. If we find the correlation we can conditionally use channelid without the user's prompt

**Comment by Strixx76** [comment]
What I have found in the API documentation about channelId is something about NVR. But could it also be that devises with two lenses/cameras uses the channelId?

Any way I have now sorted out the mess in my form. If you would like to see a suggestion to solve it you can find it here:
https://github.com/user2684/imouapi/compare/master...Strixx76:imouapi:remove-channelid

If you think that is a good solution I can make a pull request.

**Comment by user2684** [comment]
@Strixx76 I like your approach. let me see if I can refactor it together with another couple of changes I need to do in the short term and I'll come back to you. Many thanks!

**Comment by Strixx76** [comment]
Could I also the suggest that you change so that the "sensors" gets updated asynchronous?

If you change the for-loop in [ImouDevice.async_get_data](https://github.com/user2684/imouapi/blob/e559f86d8d85d18b5f3c00722f6294e264af1c7e/imouapi/device.py#L414) to use asyncio.gather() you will speed up things. That will also do that the sensors not failing will be updated. Now the loop breaks when one of the sensor fails to update. You could of course instead simply catch all exceptions here and log them like I did to be able to find out what was casing the bug.

**Comment by user2684** [comment]
Hi, closing since I have recently decided to stop actively developing the integration. For more information, have a look at https://github.com/user2684/imou_life/wiki/Status-of-the-integration
Thanks!
